typescript/cli/src/config/warp.ts

```diff
@@ -20,16 +20,38 @@ const ConnectionConfigSchema = {
 };
 
 export const WarpRouteDeployConfigSchema = z.object({
-  base: z.object({
-    type: z.literal(TokenType.native).or(z.literal(TokenType.collateral)),
-    chainName: z.string(),
-    address: ZHash.optional(),
-    isNft: z.boolean().optional(),
-    name: z.string().optional(),
-    symbol: z.string().optional(),
-    decimals: z.number().optional(),
-    ...ConnectionConfigSchema,
-  }),
+  base: z
+    .object({
+      type: z
+        .literal(TokenType.native)
+        .or(z.literal(TokenType.collateral))
+        .or(z.literal(TokenType.collateralVault)),
+      chainName: z.string(),
+      vaultAddress: ZHash.optional(),
+      address: ZHash.optional(),
+      isNft: z.boolean().optional(),
+      name: z.string().optional(),
+      symbol: z.string().optional(),
+      decimals: z.number().optional(),
+      ...ConnectionConfigSchema,
+    })
+    .refine(
+      (data) => {
+        // For collateralVault Warp Routes, ensure vaultAddress is not null or address(0).
+        if (
+          data.type === TokenType.collateralVault &&
+          (data.vaultAddress === null ||
```

Comment:
you dont want to reuse token for the vault?

Reply:
Updated